pkg/dogstatsd/server.go

```diff
@@ -466,6 +490,80 @@ func (s *Server) storeMetricStats(sample metrics.MetricSample) {
 	ms.Name = sample.Name
 	ms.Tags = strings.Join(sample.Tags, " ") // we don't want/need to share the underlying array
 	s.Debug.Stats[key] = ms
+
+	s.Debug.metricsCounts.metricChan <- struct{}{}
+}
+
+// EnableMetricsStats enables the debug mode of the DogStatsD server and start
+// the debug mainloop collecting the amount of metrics received.
+func (s *Server) EnableMetricsStats() {
+	s.Debug.Lock()
+	defer s.Debug.Unlock()
+
+	// already enabled?
+	if atomic.LoadUint64(&s.Debug.Enabled) == 1 {
```

Comment:
Yeah it's not needed but I've preferred to use the atomic method because I consider this int as an atomic int (that's how it is used, to avoid a mutex in the dogstatsd parsing hot-path) and thus I prefer to use the atomic methods anywhere else while manipulating it: it's not less clear and it let know other developers that it's better to manipulate it this way.